netket/sampler/rules/hamiltonian.py

```diff
@@ -114,3 +116,70 @@ def _choose(vp, sections, rand_vec, out, w):
         out[i] = vp[n_rand]
         w[i] = math.log(s - low_range)
         low_range = s
+
+
+@struct.dataclass
+class HamiltonianRuleJax(HamiltonianRule):
+    """
+    Rule proposing moves according to the terms in an operator.
+
+    In this case, the transition matrix is taken to be:
+
```

Comment:
need doctoring explaining this only works with jax friendly operators

Reply:
its 5 lines below